### ISSUE TITLE
UUIDs are kinda long

### ISSUE THREAD

**Issue by notriddle** [notriddle/more-interesting#2]
Potential alternatives:

* Get rid of the double ID's and just use `id INTEGER PRIMARY KEY DEFAULT RANDOM()` (that's not a lot of entropy, though)

* Use UUID's with a different encoder than the standard hex notation (base64 would shrink it down nicely).

* Generate whatever random scheme I like, doing it in Rust and just inserting it into the database at insertion time.

**Comment by NieDzejkob** [comment]
- Expose successive post ids encrypted with a block cipher. Shouldn't take more than 200 CPU cycles and provides high entropy as perceived by the user without any additional storage requirements.

**Comment by notriddle** [comment]
That does sound better. I've already got a secret key for session cookies, so could probably try to reuse that existing infrastructure.

**Comment by notriddle** [comment]
I don't think I'm going to go with the encryption method, unless I can find a better cryptosystem than what I've been able to find so far:

* I don't want to use raw AES, because that would make More Interesting a [padding oracle](https://en.wikipedia.org/wiki/Padding_oracle_attack).

* I don't want to use AES with HMAC, because that would make the URLs as long as if we had just used UUIDs. The variant included in Ring ([AEAD](https://docs.rs/ring/0.14.6/ring/aead/index.html)) also requires a nonce, which would need to be kept consistent for an individual post (otherwise the post's URL would randomly change every time) but would need to be different for each post (otherwise, padding oracle).

I think I'll start by just switching the base16 UUID over to base128. That should shrink them down to 19 characters.
